packages/grid/_modules_/grid/models/colDef/gridColDef.ts

```diff
@@ -226,18 +226,20 @@ export interface GridColumnsMeta {
   positions: number[];
 }
 
-export type GridColDefOverride<ForcedFields extends keyof GridColDef> = Omit<
-  Partial<GridColDef>,
-  ForcedFields
->;
+export interface GridGroupingColDefOverride
+  extends Omit<
+    GridColDef,
+    'editable' | 'valueSetter' | 'field' | 'preProcessEditCellProps' | 'renderEditCell'
+  > {}
 
-export type GridColDefOverrideCallback<ForcedFields extends keyof GridColDef> = (
-  params: GridColDefOverrideParams,
-) => GridColDefOverride<ForcedFields>;
+export interface GridGroupingColDefOverrideParams {
+  /**
+   * The name of the grouping algorithm currently building the grouping column.
+   */
+  treeGroupingName: string;
 
-export interface GridColDefOverrideParams {
   /**
-   * The column we are generating before the override.
+   * The base columns this column is grouping.
    */
-  colDef: GridColDef;
+  sources: GridColDef[];
```

Comment:
Should it be named "fields" and contain only the fields? Is there an use case where the full column definition needs to be present? In https://deploy-preview-3277--material-ui-x.netlify.app/components/data-grid/group-pivot/#custom-grouping-columns it only uses the field name. 

Passing the column reference opens a way for users to directly update the  column, which may cause unwanted behaviors.

Reply:
I can see edge cases where you would want to base your grouping column on other things than the field.
But we can probably start with the fields, I agree that it would be safer and would avoid some dirty mutations on the user side.